packages/eas-cli/src/project/metroConfig.ts

```diff
@@ -16,10 +16,11 @@ export async function validateMetroConfigForManagedWorkflowAsync(
   }
 
   const metroConfig = await loadConfigAsync(ctx.projectDir);
-  const hasHashAssetFilesPlugin = metroConfig.transformer?.assetPlugins?.find((plugin: string) =>
-    plugin.match(/expo-asset[/|\\]tools[/|\\]hashAssetFiles/)
-  );
-  if (!hasHashAssetFilesPlugin) {
+  
+  // This is a custom property that we inject to ensure cache invalidation between projects.
+  const isExtendingExpoMetroConfig = metroConfig.transformer.hasOwnProperty('_expoRelativeProjectRoot');
```

Comment:
do we need to keep the old logic for sdk 52 and below?

Reply:
Good point, looks like this property is actually [relatively new](https://github.com/expo/expo/commit/5ba784171ea0a7b58989414161a8fd8a1d75ad7e) - I'll add the old check back as a fallback.